frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx

```diff
@@ -19,13 +22,42 @@ const buildConfig = (name, networkType, typeParamsData): NetworkAttachmentDefini
     cniVersion: '0.3.1',
   };
 
-  _.forOwn(typeParamsData, (val, key) => {
-    config[key] = _.get(val, 'value', null);
-  });
+  let ipam = {};
+  try {
+    ipam = JSON.parse(_.get(typeParamsData, 'ipam.value', {}));
+  } catch (e) {
+    console.log(e); // eslint-disable-line no-console
+  }
+
+  if (networkType === 'cnv-bridge') {
+    const vlan = _.get(typeParamsData, 'vlanTagNum.value', '');
+    config.plugins = [
+      {
+        type: 'cnv-bridge',
+        bridge: _.get(typeParamsData, 'bridge.value', ''),
+        vlan: _.isEmpty(vlan) ? undefined : vlan,
+        ipam,
+      },
+      { type: 'cnv-tuning' },
+    ];
+  } else if (networkType === 'sriov') {
+    config.ipam = ipam;
+  }
 
   return config;
 };
 
+const getResourceName = (networkType, typeParamsData): string => {
+  let resourceName = '';
```

Comment:
since we have only `cnv-bridge` and `sriov` then this can be `const`

```suggestion
  const resourceName = networkType === 'cnv-bridge' ?  _.get(typeParamsData, 'bridge.value', '') : .get(typeParamsData, 'resourceName.value', '');
```

Reply:
Done! Premature optimization gets me again. lol